pkg/apis/monitoring/v1/types.go

```diff
@@ -1816,22 +1816,34 @@ type PrometheusRuleSpec struct {
 // or 'abort'.  More info: https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#partial-response
 // +k8s:openapi-gen=true
 type RuleGroup struct {
-	Name                    string `json:"name"`
-	Interval                string `json:"interval,omitempty"`
-	Rules                   []Rule `json:"rules"`
+	// Name of the rule group.
+	Name string `json:"name"`
+	// Interval determines how often rules in the group are evaluated.
+	Interval string `json:"interval,omitempty"`
+	// Alerting or Recording Rules
+	Rules []Rule `json:"rules"`
+	// PartialResponseStrategy is only used by ThanosRuler
```

Comment:
It might be good to remove L1814-1816 to avoid repetition.

```suggestion
	// partialResponseStrategy is only relevant when the PrometheusRule object is deployed to Thanos Ruler.
	// This field is ignored when the object is deployed to Prometheus.
```